addons/l10n_it_edi/models/res_partner.py

```diff
@@ -31,21 +32,135 @@ class ResPartner(models.Model):
             "Destination Code must have between 6 and 7 characters."),
     ]
 
-    @api.model
-    def _l10n_it_normalize_codice_fiscale(self, codice):
-        if codice and re.match(r'^IT[0-9]{11}$', codice):
-            return codice[2:13]
-        return codice
+    def _l10n_it_edi_get_values(self):
+        """ Generates all partner values needed by l10n_it_edi XML export """
+        if not self:
+            return False
+
+        self.ensure_one()
+        partner = self
+        class NormalizedPartnerValues(UserDict):
```

Comment:
We can remove the NormalizedPartnerValues class here and use the seller vs seller_partner in the qweb view (as we already have them).

Reply:
Done... I have some considerations imho:
1) seller (buyer sender) can both be a company or a partner so when we call `is_company` we have anyway to distinguish
2) the `Sede` template with the addresses needs both, so we pass 2 variables
3) It's not clear at all by looking at the xml if there is an error. Having one variable with both partner and company data would be useful
4) We are way well beyond the point of a normal "XML View" export. QWeb works well when it presents data that it's almost strictly bound to Odoo data. This is data integration, it's really something else, I talk from years of experience. Generally, for export, there are export models, or export SQL tables.
This "lightweight" way we have is more like inline PHP than Python -- even if you would say that the object way is Java.